nixos/modules/services/networking/firewall.nix

```diff
@@ -177,6 +177,21 @@ let
       ) cfg.allowedUDPPortRanges
     }
 
+    # Accept packets on the extra allowed host+port combination.
+    ${concatMapStrings (allowed:
+      if (allowed.useIPv4 || allowed.usedIPv6) then (
```

Comment:
`allowed.useIpv6` not `allowed.usedIpv6`. If you use an enum for the address family you can drop this `if` clause.